src/charm.py

```diff
@@ -826,6 +857,11 @@ def _on_database_storage_detaching(self, _) -> None:
         # Inform other hooks of current status
         self.unit_peer_data["unit-status"] = "removing"
 
+        if self.unit.is_leader():
+            # Update 'units-added-to-cluster' counter in the peer relation databag
+            units = int(self.app_peer_data.get("units-added-to-cluster", 1))
+            self.app_peer_data["units-added-to-cluster"] = str(units - 1)
```

Comment:
should we also have this update to `units-added-to-cluster` on the `peer-relation-departed` event? wouldnt we like to decrement this value if non-leader units are scaled down?

or is this happening on `update-status` (in particular `_set_app_status`)? if this is the case, would there be inconsistency issues -> there are 3 units, if another unit departed and then the leader unit departed, `units-added-to-cluster` will be 2 instead of 1 (if `storage-detaching` runs on leader before `update-status`)